authority/ssh.go

```diff
@@ -146,22 +146,24 @@ func (a *Authority) GetSSHBastion(ctx context.Context, user, hostname string) (*
 }
 
 // SignSSH creates a signed SSH certificate with the given public key and options.
-func (a *Authority) SignSSH(_ context.Context, key ssh.PublicKey, opts provisioner.SignSSHOptions, signOpts ...provisioner.SignOption) (*ssh.Certificate, error) {
+func (a *Authority) SignSSH(_ context.Context, key ssh.PublicKey, opts provisioner.SignSSHOptions, signOpts ...provisioner.SignOption) (cert *ssh.Certificate, err error) {
+	var prov provisioner.Interface
+	defer a.incrProvisionerCounter(&prov, &err, Meter.SSHSigned)
```

Comment:
We can also clean this up by moving the signing code to internal methods and getting the error from the return values.

Reply:
We're already setting the error to `nil` when `storeCertificate` fails and the `errors.Is(err,  db.ErrNotImplemented)`.

If we base our decision to instrument based on whether the returned chain is nil, then we lose the ability (in other places, outside the scope of open source) to instrument in more detail (perhaps according to the underlying type of error).

Reply:
Whether `(*Authority).storeCertificate` should return a `nil` error, where it now returns `db.ErrNotImplemented` (so that we can avoid checking for the particular exception), is something to consider, for the future.

Reply:
In fact, I've changed the signature of meter functions to directly accept the provisioner & the error.